qiskit_machine_learning/algorithms/inference/qbayesian.py

```diff
@@ -62,7 +66,8 @@ def __init__(
         *,
         limit: int = 10,
         threshold: float = 0.9,
-        sampler: BaseSampler | None = None,
+        sampler: BaseSampler | BaseSamplerV2 | None = None,
```

Comment:
Below there is this code

```
        if sampler is None:
            sampler = Sampler()
```

which sets a default instance should the user not provide one. Should we already switch this, and other places that do this, over to using V2 now? If there is some behavorial difference that manifests in a different outcome and it should be left like this for now, maybe it should issue a deprecation warning saying that the default will be changed to V2 instance at the same time the V1 primitive support in ML is removed.

Reply:
Just updated with this!